mono/metadata/unity-liveness.c

```diff
@@ -292,6 +390,73 @@ static gboolean mono_traverse_object_internal(MonoObject *object, gboolean isStr
 				MonoVTable *vtable = NULL;
 				mono_field_get_value_internal(object, field, &val);
 				added_objects |= mono_add_process_object(val, state);
+				validate_object_value(val, field->type);
```

Comment:
This has been added to existing code. Is that correct? Is it adding cost at runtime?